### PR TITLE
Clean up and improve automod

### PR DESCRIPTION
**Pull request by zMoooooritz**
Does build successfully and looked as if it is still running as it is supposed to.

If anyone has the server and time capacity to test it properly I would be thankful

### CODE REVIEW

File: config/default_config.yml [hunk 1]
```diff
@@ -116,35 +116,32 @@ GTX:
...
   # disabled the kick and have 3 punishes configured, once the 3 punishes are 
   # through that's it, nothing will happen anymore for that squad even if they still dont have an officer. 
   # 
   # Note 2: punishes and kicks are tracked individual per player, not at the squad level.
   # So if a player joins a squad that already received 3 punishes and will be kicked next, the new player will 
-  # start at the first punish regardless (so he won't be kick), howver he won't have a warning broadcast message as those are issued and tracked per squad)
-  enabled: false
+  # start at the first punish regardless (so he won't be kick)
+  # Set to true to enable the auto moderation of squads without officers
+  enabled: true
```
Comment (MarechJ) [inline config/default_config.yml:128]: hmm I'd not change that default just yet

File: rcon/squad_automod/automod.py [hunk 2]
```diff
@@ -63,154 +65,151 @@ def is_time(times: List[datetime], interval_seconds: int):
...
+    aplayer: APlayer
 ):
+    warnings = watch_status.warned.setdefault(aplayer.name, [])
+
+    if len(warnings) == 0:
+        logger.debug("Noted but won't warn: %s in %s", aplayer.short_repr(), squad_name)
+        warnings.append(datetime.now())
+        return PunishStepState.WAIT
+
     if config.number_of_warning == 0:
```
Comment (MarechJ) [inline rcon/squad_automod/automod.py:87]: You should move that before the previous block

File: rcon/squad_automod/automod.py [hunk 3]
```diff
@@ -63,154 +65,151 @@ def is_time(times: List[datetime], interval_seconds: int):
...
 
     if len(squad["players"]) < config.min_squad_players_for_punish:
         logger.debug("Squad %s below min player count for punish", squad_name)
-        return PunishStepState.wait
+        return PunishStepState.WAIT
 
     if (
-        int(player["level"]) <= config.immuned_level_up_to
-        or player["role"] in config.immuned_roles
+        aplayer.lvl <= config.immuned_level_up_to
```
Comment (MarechJ) [inline rcon/squad_automod/automod.py:138]: I wouldn't remove the type case here. I'm not sure the playerinfo method is casting it and APlayer doesn't either

File: rcon/squad_automod/automod.py [hunk 4]
```diff
@@ -63,154 +65,151 @@ def is_time(times: List[datetime], interval_seconds: int):
...
 
     if len(squad["players"]) < config.min_squad_players_for_kick:
         logger.debug("Squad %s below min player count for punish", squad_name)
-        return PunishStepState.wait
+        return PunishStepState.WAIT
 
     if (
-        int(player["level"]) <= config.immuned_level_up_to
-        or player["role"] in config.immuned_roles
+        aplayer.lvl <= config.immuned_level_up_to
```
Comment (MarechJ) [inline rcon/squad_automod/automod.py:194]: Same here, typecast for safety

File: rcon/squad_automod/models.py [hunk 5]
```diff
@@ -65,56 +78,31 @@ class NoLeaderConfig:
...
+            f"(name={self.name}, lvl={self.lvl}, role={self.role})"
+        )
 
 
 @dataclass
 class PunitionsToApply:
     warning: List[APlayer] = field(default_factory=list)
     punish: List[APlayer] = field(default_factory=list)
     kick: List[APlayer] = field(default_factory=list)
-    squads_state: List[ASquad] = field(default_factory=list)
```
Comment (MarechJ) [inline rcon/squad_automod/models.py:87]: I think we should keep that for debug purpose until it's stable

**Review by MarechJ — COMMENTED** [review]
Do the unittest pass?

File: config/default_config.yml (hunk 1 above)
Comment (zMoooooritz) [reply config/default_config.yml:128]: Yes good catch, residue of testing

File: rcon/squad_automod/automod.py (hunk 2 above)
Comment (zMoooooritz) [reply rcon/squad_automod/automod.py:87]: Did reorder those blocks to have the grace period for all (warn, punish, kick) without implementing it in each method separately.
If there is a more elegant way let me know

**Comment by zMoooooritz** [comment]
> Do the unittest pass?

Yes now they do.
The new solution does add some computational overhead but I think this is pretty clean and nicely configurable

**Review by MarechJ — APPROVED** [review]
lgtm you can merge

**Pull request merged**